### ISSUE TITLE
Add zsh completions to homebrew formula

### ISSUE THREAD

**Issue by sandr01d** [wfxr/forgit#332]
## Check list

- [X] I have read through the [README](https://github.com/wfxr/forgit/blob/master/README.md)
- [X] I have the latest version of forgit
- [X] I have searched through the existing issues

Noticed the [homebrew formula](https://github.com/Homebrew/homebrew-core/blob/64ba73f1b2f22f88a1d51c0cbdcf7f80cc58c9d1/Formula/f/forgit.rb) installs completions for bash, but not zsh. As I don't use homebrew myself, I currently don't have a setup to test this, maybe someone else is interested in doing this?

**Comment by cjappl** [comment]
Pinging @carlfriedrich as the original master of the homebrew stuff.

I'd love to learn how to run and test this stuff, if you want to document it somewhere I can help next time.

**Comment by carlfriedrich** [comment]
@sandr01d I actually set up forgit in homebrew once and never updated it. I was planning to add a GitHub action to update it automatically with each release, just like we do with AUR, but I didn't have time for it, yet, and furthermore I don't have access to a Mac in order to test the homebrew stuff.
@cjappl Homebrew is actually well documented. I had to check it out myself in order to implement forgit in there, because I haven't used it before, so you might give it a try as well?

**Comment by cjappl** [comment]
Happy to investigate this. I think I can also try to get it automated. it'd be nice to have our freshest up all the time.

**Comment by cjappl** [comment]
Alright, there are multiple rounds we have to go through here, as our completions aren't exactly working still. I think we need to change them on our side of the fence.

To get the ball rolling, I submitted a cleanup PR:
https://github.com/Homebrew/homebrew-core/pull/161025

This was mostly to get my feet wet in the homebrew world so I could do the meatier change of fixing the completions. Check it out if you get a chance.

**Comment by carlfriedrich** [comment]
@cjappl Great, thanks for taking this over!

**Comment by cjappl** [comment]
@carlfriedrich could you walk me (an idiot, who doesn't understand zsh) how do use our completions?

Like very explicitly:
1. Copy file to x
2. run this command
3. `ga <tab>` should look like ...

I cannot get them to work, but I 100% expect user error. Please be as explicit as possible, as I don't have any zsh experience.

**Comment by carlfriedrich** [comment]
Pinging @sandr01d who implemented the zsh completions.

**Comment by sandr01d** [comment]
@cjappl sounds like compsys (zsh's completion system) is not initialized. Let me walk you through the setup and try to provide a bit of background information that might come in handy.
For completions in zsh to work, compsys has to be explicitly loaded and initialized in your .zshrc. The most common way to do so is to place the following snippet somewhere at the top of your .zshrc, before you load any plugins:

```sh
autoload -Uz compinit
compinit
```

`$fpath` is a variable similar to the `$PATH` environment variable, but specific to zsh completion functions. Compinit will scan files in in the directories present in `$fpath` and dump a bit of information about the functions defined in these files in ~/.zcompdump. In case you want to check whether compinit correctly finds your completion functions, this is a good place to start. The first time you try to use a completion for a command, compsys will check whether a completion function has been noted in your compdump and, if so, will load and cache the completion function from the corresponding completion file. So you have to place *completions/_git-forgit* in a directory that is listed in your `$fpath` as you seem to have done correctly. */opt/homebrew/share/zsh/site-functions* is not part of `$fpath` by default. I suspect that homebrew is adding this directory to the `$fpath` for plugins installed with homebew. I think this should work fine, but in case you're having issues you might want to test one of the other directories from your `$fpath`, */usr/local/share/zsh/site-functions* would usually be a good start.
Of course you also have to source forgit as usual from your .zshrc, but I'm suspecting you already do so. At this point you should get completions for forgit as a git subcommand, e.g.

```
git forgit <TAB>
```
should print a list of available commands.
For the completions to also work with forgits aliases and functions, you have to source *completions/git-forgit.zsh* **after** sourcing forgit, e.g.:

```sh
source /path/to/forgit.plugin.zsh
source /path/to/completions/git-forgit.zsh
```

Let me know whether things are working out for you and whether anything I've provided is unclear and I will do my best to help sort it out.
In general the man page for [zshcompsys](https://linux.die.net/man/1/zshcompsys) and sometimes [zshmisc](https://linux.die.net/man/1/zshmisc) are usually a good resource.

**Comment by cjappl** [comment]
Thank you. Very helpful.

Could you explain the difference between our two files:
https://github.com/wfxr/forgit/blob/master/completions/_git-forgit

and
https://github.com/wfxr/forgit/blob/master/completions/git-forgit.zsh

Which file should be sourced? are the names important? Do they both need to be on `$fpath`? (again, any additional info greatly appreciated)

**Comment by sandr01d** [comment]
@cjappl Can this issue be closed? Looks to me like #340 and https://github.com/Homebrew/homebrew-core/pull/164702 solve this.

**Comment by cjappl** [comment]
Seems good to me! Thanks for noticing this was still open.
